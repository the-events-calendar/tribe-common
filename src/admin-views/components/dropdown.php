<?php
/**
 * View: Common Dropdown Input.
 *
 * Override this template in your own theme by creating a file at:
 * [your-theme]/tribe/admin-views/components/dropdown.php
 *
 * See more documentation about our views templating system.
 *
 * @link    https://evnt.is/1aiy
 *
 * @since 1.0.0
 *
 * @version 1.0.0
 *
 * @var string               $label          Label for the dropdown input.
 * @var string               $id             ID of the dropdown input.
 * @var array<string,string> $classes_wrap   An array of classes for the dropdown wrap.
 * @var array<string,string> $classes_label  An array of classes for the label.
 * @var array<string,string> $classes_select An array of classes for the dropdown input.
 * @var string               $name           Name attribute for the dropdown input.
 * @var string|int           $selected       The selected option id.
 * @var array<string,string> $attrs          Associative array of attributes of the dropdown.
 * @var array<string,string> $wrap_attrs     Associative array of attributes of the dropdown wrap.
 * @var array<string,string> $tooltip        An optional associative array containing information to display in a tooltip.
 */

$wrap_classes = [ 'tec-settings-control', 'tec-settings-control__dropdown-wrap' ];
if ( ! empty( $classes_wrap ) ) {
	$wrap_classes = array_merge( $wrap_classes, $classes_wrap );
}

$label_classes = [ 'tec-settings-control__label' ];
if ( ! empty( $classes_label ) ) {
	$label_classes = array_merge( $label_classes, $classes_label );
}

$select_classes = [ 'tribe-dropdown', 'tec-settings-control__dropdown' ];
if ( ! empty( $classes_select ) ) {
	$select_classes = array_merge( $select_classes, $classes_select );
}

if ( empty( $wrap_attrs ) ) {
	$wrap_attrs = [];
}
?>
<div
	<?php tec_classes( $wrap_classes ); ?>
	<?php tribe_attributes( $wrap_attrs ); ?>
>
	<label
		<?php tec_classes( $label_classes ); ?>
		for="<?php echo esc_attr( $id ); ?>"
	>
		<?php echo esc_html( $label ); ?>
	</label>
	<?php
	// Move opening and closing php tags next to html tags to prevent whitespace and alignment issues.
	if ( ! empty( $tooltip['message'] ) ) {
		$this->template( 'components/tooltip', $tooltip );
	}
	?>
	<select
		id="<?php echo esc_attr( $id ); ?>"
		name="<?php echo esc_attr( $name ); ?>"
		<?php tec_classes( $select_classes ); ?>
		value="<?php echo esc_attr( $selected ); ?>"
		style="width: 100%;" <?php /* This is required for selectWoo styling to prevent select box overflow */ ?>
		<?php tribe_attributes( $attrs ); ?>
	>
	</select>
</div>
