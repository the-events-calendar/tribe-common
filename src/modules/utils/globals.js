/**
 * @todo: handle globals in a better way
 */

export const get = ( key, defaultValue ) => window[ key ] || defaultValue;
export const google = () => get( 'google' );
export const wpApi = wp.api;
export const wpApiRequest = wp.apiRequest;
export const wpComponents = wp.components;
export const wpData = wp.data;
export const wpEditor = wp.blockEditor || wp.editor;
export const wpHooks = wp.hooks;

export const wpDataSelectCoreEditor = () => wpData.select( 'core/block-editor' ) ||
	wpData.select( 'core/editor' );

export const IconButton = wpComponents.Button || wpComponents.IconButton;

// Localized Config
export const config = () => get( 'tribe_editor_config', {} );

// Common
export const common = () => config().common || {};
export const adminUrl = () => common().adminUrl || '';
export const rest = () => common().rest || {};
export const restNonce = () => rest().nonce || {};
export const dateSettings = () => common().dateSettings || {};
export const editorConstants = () => common().constants || {};
export const list = () => ( {
	countries: common().countries || {},
	us_states: common().usStates || {},
} );

// TEC
export const tec = () => config().events || {};
export const editor = () => tec().editor || {};
export const settings = () => tec().settings || {};
export const mapsAPI = () => tec().googleMap || {};
export const priceSettings = () => tec().priceSettings || {};
export const tecDateSettings = () => tec().dateSettings || {};
export const timezoneHtml = () => tec().timezoneHTML || '';
export const defaultTimes = () => tec().defaultTimes || {};
export const timezone = () => tec().timeZone || {};

// PRO
export const pro = () => config().eventsPRO || {};
export const editorDefaults = () => pro().defaults || {};

// Tickets
export const tickets = () => config().tickets || {};

// Tickets Plus
export const ticketsPlus = () => config().ticketsPlus || {};
export const iacVars = () => ticketsPlus().iacVars || {};

// Post Objects
export const postObjects = () => config().post_objects || {};
export const post = () => config().post || {};

// Blocks
export const blocks = () => config().blocks || {};
