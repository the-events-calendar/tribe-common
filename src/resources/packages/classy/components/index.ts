export { CenteredSpinner } from './CenteredSpinner';
export { CurrencyInput } from './CurrencyInput';
export { CurrencySelector } from './CurrencySelector';
export { DatePicker } from './DatePicker';
export { EndSelector } from './EndSelector';
export { ErrorBoundary } from './ErrorBoundary';
export * from './Icons';
export { LabeledInput } from './LabeledInput';
export { TimePicker } from './TimePicker';
export { TimeZone } from './TimeZone';
export { TinyMceEditor } from './TinyMceEditor';
export { StartSelector } from './StartSelector';

// Note: the Classy application is not exported by design, only components are.
