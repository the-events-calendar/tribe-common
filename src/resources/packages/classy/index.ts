import { whenEditorIsReady } from './functions/whenEditorIsReady';
import { hideInserterToggle, hideZoomOutButton, hideSidebarBlockTab } from './functions/editorModifications';
import {
	initApp as initClassyApp,
	insertElement as insertClassyElement,
} from './functions/classy';
import { getLocalizedData, getSettings } from './localizedData';
import './style.pcss';

whenEditorIsReady().then( () => {
	hideZoomOutButton();
	hideInserterToggle();
	hideSidebarBlockTab();
	initClassyApp();
	insertClassyElement();
} );

// Re-exports that will appear under `window.tec.common.classy.<re-export>`.
export * as components from './components';
export * as fields from './fields';
export * as functions from './functions';
export * as store from './store';

/*
 * Re-export localized data accessors and not the localized data object directly.
 * Packages outside of this will be able to access the localized data in one of two ways:
 *
 * Recommended:
 * - import {getLocalizedData, getSettings} from '@tec/common/classy/localizedData';
 *
 * Not ideal but still possible:
 * - const {getLocalizedData, getSettings} = window.tec.common.classy.localizedData;
 */
export const localizedData = {
	getLocalizedData,
	getSettings,
};
