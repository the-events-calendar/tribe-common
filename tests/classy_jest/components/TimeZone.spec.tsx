import * as React from 'react';
import { render, waitFor } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import { beforeEach, describe, expect, it, jest } from '@jest/globals';
import { TimeZone } from '../../../src/resources/packages/classy/components';

describe( 'TimeZone Component', () => {
	const defaultProps = {
		onTimezoneChange: jest.fn(),
		timezone: 'Europe/Paris',
	};

	const popoverSelector = '.classy-component__popover--timezone';
	// The `is-positioned` class is added when the popover is positioned, showing and transitions are completed.
	const positionedPopoverSelector = `${ popoverSelector }.is-positioned`;
	const timezoneButtonSelector = '.is-link.classy-field__timezone-value';

	beforeEach( () => {
		jest.resetModules();
		jest.clearAllMocks();
	} );

	it( 'renders correctly with default props', async () => {
		const user = userEvent.setup();
		const { container } = render( <TimeZone { ...defaultProps } /> );

		// To start, there should not be a timezone selection popover open.
		let popover = document.querySelector( popoverSelector );
		expect( popover ).toBeNull();

		expect( container ).toMatchSnapshot();

		const button = container.querySelector( timezoneButtonSelector ) as Element;
		expect( button ).not.toBeNull();

		// Click the timezone selection button to open the timezone selection popover.
		await user.click( button );

		// Wait for the popover to be positioned.
		await waitFor( () => {
			expect( document.querySelector( positionedPopoverSelector ) ).not.toBeNull();
		} );

		popover = document.querySelector( positionedPopoverSelector ) as Element;
		expect( popover ).not.toBeNull();
		expect( popover ).toMatchSnapshot();
	} );

	it( 'renders the UTC timezone choice if the timezone string is UTC', async () => {
		const user = userEvent.setup();
		const { container } = render( <TimeZone { ...defaultProps } timezone="UTC+1" /> );

		// To start, there should not be a timezone selection popover open.
		let popover = document.querySelector( popoverSelector );
		expect( popover ).toBeNull();

		expect( container ).toMatchSnapshot();

		const button = container.querySelector( timezoneButtonSelector ) as Element;
		expect( button ).not.toBeNull();

		// Click the timezone selection button to open the timezone selection popover.
		await user.click( button );

		// Wait for the popover to be positioned.
		await waitFor( () => {
			expect( document.querySelector( positionedPopoverSelector ) ).not.toBeNull();
		} );

		popover = document.querySelector( positionedPopoverSelector ) as Element;
		expect( popover ).not.toBeNull();
		expect( popover ).toMatchSnapshot();
	} );

	it( 'allows selecting a new timezone', async () => {
		const user = userEvent.setup();
		const { container, asFragment } = render( <TimeZone { ...defaultProps } /> );

		// To start, there should not be a timezone selection popover open.
		let popover = document.querySelector( popoverSelector );
		expect( popover ).toBeNull();

		expect( container ).toMatchSnapshot();

		const button = container.querySelector( timezoneButtonSelector );
		expect( button ).toBeDefined();

		// Click the timezone selection button to open the timezone selection popover.
		await user.click( button as Element );

		// Wait for the popover to be positioned.
		await waitFor( () => {
			expect( document.querySelector( positionedPopoverSelector ) ).not.toBeNull();
		} );

		// In the popover select Abidjan (Africa/Abidjan).
		popover = document.querySelector( popoverSelector );
		const timezoneOptions = ( popover as Element ).querySelector(
			'.classy-component__popover-input--timezone select'
		) as HTMLSelectElement;

		expect( timezoneOptions ).toBeDefined();
		// There should be an Africa/Abidjan option in the select element.
		expect( timezoneOptions.querySelector( 'option[value="Africa/Abidjan"]' ) ).not.toBeNull();

		await user.selectOptions( timezoneOptions, [ 'Africa/Abidjan' ] );

		// The popover should have closed and the new timezone should have been selected.
		popover = document.querySelector( popoverSelector );
		expect( popover ).toBeNull();
		expect( defaultProps.onTimezoneChange ).toHaveBeenCalledWith( 'Africa/Abidjan' );
	} );
} );
