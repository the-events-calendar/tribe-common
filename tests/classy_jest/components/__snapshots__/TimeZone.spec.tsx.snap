// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeZone Component allows selecting a new timezone 1`] = `
<div>
  <div
    class="classy-field__control classy-field__control--timezone"
  >
    <span
      class="classy-field__timezone-label"
    >
      Time Zone:
    </span>
     
    <button
      class="components-button is-link is-link--dark classy-field__timezone-value is-link"
      type="button"
    >
      Europe/Paris
    </button>
  </div>
</div>
`;

exports[`TimeZone Component renders correctly with default props 1`] = `
<div>
  <div
    class="classy-field__control classy-field__control--timezone"
  >
    <span
      class="classy-field__timezone-label"
    >
      Time Zone:
    </span>
     
    <button
      class="components-button is-link is-link--dark classy-field__timezone-value is-link"
      type="button"
    >
      Europe/Paris
    </button>
  </div>
</div>
`;

exports[`TimeZone Component renders correctly with default props 2`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 13px;
  box-sizing: border-box;
}

.emotion-0 *,
.emotion-0 *::before,
.emotion-0 *::after {
  box-sizing: inherit;
}

.components-panel__row .emotion-2 {
  margin-bottom: inherit;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: calc(4px * 2);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  box-sizing: border-box;
  position: relative;
  border-radius: 2px;
  padding-top: 0;
  color: var(--wp-components-color-foreground, #1e1e1e);
  cursor: pointer;
}

.emotion-6>* {
  min-height: 0;
}

.emotion-6:focus-within:not( :has( :is( .em5sgkm8, .emotion-13 ):focus-within ) ) .emotion-20 {
  border-color: var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
  box-shadow: 0 0 0 0.5px var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.emotion-8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  background-color: var(--wp-components-color-background, #fff);
  width: 100%;
}

.emotion-10.emotion-10.emotion-10 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: transparent;
  box-sizing: border-box;
  border: none;
  box-shadow: none!important;
  color: currentColor;
  cursor: inherit;
  display: block;
  font-family: inherit;
  margin: 0;
  width: 100%;
  max-width: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  font-size: 16px;
  height: 40px;
  min-height: 40px;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 12px;
  padding-right: 30px;
  overflow: hidden;
}

@media ( min-width: 600px ) {
  .emotion-10.emotion-10.emotion-10 {
    font-size: 13px;
  }
}

.emotion-12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-15 {
  -webkit-padding-end: 12px;
  padding-inline-end: 12px;
  position: absolute;
  pointer-events: none;
  right: 0;
}

.emotion-17 {
  -webkit-margin-end: calc(4px * -1);
  margin-inline-end: calc(4px * -1);
  line-height: 0;
}

.emotion-17 path {
  fill: currentColor;
}

.emotion-19.emotion-19.emotion-19 {
  box-sizing: border-box;
  border-color: var(--wp-components-color-gray-600, #949494);
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  bottom: 0;
  left: 0;
  margin: 0;
  padding: 0;
  pointer-events: none;
  position: absolute;
  right: 0;
  top: 0;
  padding-left: 2px;
}

<div
  class="components-popover classy-component__popover classy-component__popover--choice classy-component__popover--timezone is-positioned"
  data-wp-c16t="true"
  data-wp-component="Popover"
  style="position: absolute; top: 0px; left: 0px; opacity: 1; transform: translateY(4px); transform-origin: 100% 0% 0;"
  tabindex="-1"
>
  <div
    class="components-popover__content"
    style="max-height: 0px; overflow: auto;"
  >
    <div
      class="classy-component__popover-content"
    >
      <button
        class="components-button classy-component__popover-close is-link"
        type="button"
      >
        <span
          class="classy-icon classy-icon--close"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 13.0602L15.7123 16.7725L16.773 15.7118L13.0607 11.9995L16.773 8.28723L15.7123 7.22657L12 10.9389L8.28771 7.22656L7.22705 8.28722L10.9394 11.9995L7.22706 15.7119L8.28772 16.7725L12 13.0602Z"
              fill="#1E1E1E"
            />
          </svg>
        </span>
      </button>
      <h4
        class="classy-component__popover-title"
      >
        Event Time Zone
      </h4>
      <p
        class="classy-component__popover-description"
      >
        Choose a different time zone than your default for this event.
      </p>
      <div
        class="components-base-control emotion-0 emotion-1"
      >
        <div
          class="components-base-control__field emotion-2 emotion-3"
        >
          <div
            class="components-flex components-input-base components-select-control classy-component__popover-input classy-component__popover-input--select classy-component__popover-input--timezone emotion-4 emotion-5 emotion-6 emotion-7"
            data-wp-c16t="true"
            data-wp-component="InputBase"
          >
            <div
              class="components-input-control__container emotion-8 emotion-9"
            >
              <select
                class="components-select-control__input emotion-10 emotion-11"
                id="inspector-select-control-0"
              >
                <optgroup
                  label="Africa"
                >
                  <option
                    value="Africa/Abidjan"
                  >
                    Abidjan
                  </option>
                </optgroup>
                <optgroup
                  label="Europe"
                >
                  <option
                    value="Paris"
                  >
                    Paris
                  </option>
                </optgroup>
                <optgroup
                  label="North America"
                >
                  <option
                    value="America/New_York"
                  >
                    New York
                  </option>
                </optgroup>
              </select>
              <span
                class="components-input-control__suffix emotion-12 emotion-13"
              >
                <div
                  class="components-input-control-suffix-wrapper emotion-14 emotion-15 emotion-16"
                  data-wp-c16t="true"
                  data-wp-component="InputControlSuffixWrapper"
                >
                  <div
                    class="emotion-17 emotion-18"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      height="18"
                      viewBox="0 0 24 24"
                      width="18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
                      />
                    </svg>
                  </div>
                </div>
              </span>
              <div
                aria-hidden="true"
                class="components-input-control__backdrop emotion-19 emotion-20"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TimeZone Component renders the UTC timezone choice if the timezone string is UTC 1`] = `
<div>
  <div
    class="classy-field__control classy-field__control--timezone"
  >
    <span
      class="classy-field__timezone-label"
    >
      Time Zone:
    </span>
     
    <button
      class="components-button is-link is-link--dark classy-field__timezone-value is-link"
      type="button"
    >
      UTC+1
    </button>
  </div>
</div>
`;

exports[`TimeZone Component renders the UTC timezone choice if the timezone string is UTC 2`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 13px;
  box-sizing: border-box;
}

.emotion-0 *,
.emotion-0 *::before,
.emotion-0 *::after {
  box-sizing: inherit;
}

.components-panel__row .emotion-2 {
  margin-bottom: inherit;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: calc(4px * 2);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  box-sizing: border-box;
  position: relative;
  border-radius: 2px;
  padding-top: 0;
  color: var(--wp-components-color-foreground, #1e1e1e);
  cursor: pointer;
}

.emotion-6>* {
  min-height: 0;
}

.emotion-6:focus-within:not( :has( :is( .em5sgkm8, .emotion-13 ):focus-within ) ) .emotion-20 {
  border-color: var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
  box-shadow: 0 0 0 0.5px var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.emotion-8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  background-color: var(--wp-components-color-background, #fff);
  width: 100%;
}

.emotion-10.emotion-10.emotion-10 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: transparent;
  box-sizing: border-box;
  border: none;
  box-shadow: none!important;
  color: currentColor;
  cursor: inherit;
  display: block;
  font-family: inherit;
  margin: 0;
  width: 100%;
  max-width: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  font-size: 16px;
  height: 40px;
  min-height: 40px;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 12px;
  padding-right: 30px;
  overflow: hidden;
}

@media ( min-width: 600px ) {
  .emotion-10.emotion-10.emotion-10 {
    font-size: 13px;
  }
}

.emotion-12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-15 {
  -webkit-padding-end: 12px;
  padding-inline-end: 12px;
  position: absolute;
  pointer-events: none;
  right: 0;
}

.emotion-17 {
  -webkit-margin-end: calc(4px * -1);
  margin-inline-end: calc(4px * -1);
  line-height: 0;
}

.emotion-17 path {
  fill: currentColor;
}

.emotion-19.emotion-19.emotion-19 {
  box-sizing: border-box;
  border-color: var(--wp-components-color-gray-600, #949494);
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  bottom: 0;
  left: 0;
  margin: 0;
  padding: 0;
  pointer-events: none;
  position: absolute;
  right: 0;
  top: 0;
  padding-left: 2px;
}

<div
  class="components-popover classy-component__popover classy-component__popover--choice classy-component__popover--timezone is-positioned"
  data-wp-c16t="true"
  data-wp-component="Popover"
  style="position: absolute; top: 0px; left: 0px; opacity: 1; transform: translateY(4px); transform-origin: 100% 0% 0;"
  tabindex="-1"
>
  <div
    class="components-popover__content"
    style="max-height: 0px; overflow: auto;"
  >
    <div
      class="classy-component__popover-content"
    >
      <button
        class="components-button classy-component__popover-close is-link"
        type="button"
      >
        <span
          class="classy-icon classy-icon--close"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 13.0602L15.7123 16.7725L16.773 15.7118L13.0607 11.9995L16.773 8.28723L15.7123 7.22657L12 10.9389L8.28771 7.22656L7.22705 8.28722L10.9394 11.9995L7.22706 15.7119L8.28772 16.7725L12 13.0602Z"
              fill="#1E1E1E"
            />
          </svg>
        </span>
      </button>
      <h4
        class="classy-component__popover-title"
      >
        Event Time Zone
      </h4>
      <p
        class="classy-component__popover-description"
      >
        Choose a different time zone than your default for this event.
      </p>
      <div
        class="components-base-control emotion-0 emotion-1"
      >
        <div
          class="components-base-control__field emotion-2 emotion-3"
        >
          <div
            class="components-flex components-input-base components-select-control classy-component__popover-input classy-component__popover-input--select classy-component__popover-input--timezone emotion-4 emotion-5 emotion-6 emotion-7"
            data-wp-c16t="true"
            data-wp-component="InputBase"
          >
            <div
              class="components-input-control__container emotion-8 emotion-9"
            >
              <select
                class="components-select-control__input emotion-10 emotion-11"
                id="inspector-select-control-1"
              >
                <optgroup
                  label="Africa"
                >
                  <option
                    value="Africa/Abidjan"
                  >
                    Abidjan
                  </option>
                </optgroup>
                <optgroup
                  label="Europe"
                >
                  <option
                    value="Paris"
                  >
                    Paris
                  </option>
                </optgroup>
                <optgroup
                  label="North America"
                >
                  <option
                    value="America/New_York"
                  >
                    New York
                  </option>
                </optgroup>
                <optgroup
                  label="UTC"
                >
                  <option
                    value="UTC+0"
                  >
                    UTC
                  </option>
                </optgroup>
              </select>
              <span
                class="components-input-control__suffix emotion-12 emotion-13"
              >
                <div
                  class="components-input-control-suffix-wrapper emotion-14 emotion-15 emotion-16"
                  data-wp-c16t="true"
                  data-wp-component="InputControlSuffixWrapper"
                >
                  <div
                    class="emotion-17 emotion-18"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      height="18"
                      viewBox="0 0 24 24"
                      width="18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
                      />
                    </svg>
                  </div>
                </div>
              </span>
              <div
                aria-hidden="true"
                class="components-input-control__backdrop emotion-19 emotion-20"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
